lib/migration.js

```diff
@@ -134,7 +136,69 @@ Migration.prototype.sqlFileLoaderTemplate = function() {
     "};",
     "",
     "exports.down = function(db, callback) {",
-    "  var filePath = path.join(__dirname + '/sqls/"+this.name.replace('.js', '')+"-down.sql');",
+    "  var filePath = path.join(__dirname + '/sqls/" + this.name.replace(
+      '.js', '') + "-down.sql');",
+    "  fs.readFile(filePath, {encoding: 'utf-8'}, function(err,data){",
+    "    if (err) return callback(err);",
+    "    console.log('received data: ' + data);",
+    "",
+    "    db.runSql(data, function(err) {",
+    "      if (err) return callback(err);",
+    "      callback();",
+    "    });",
+    "  });",
+    "};",
+    ""
+  ].join("\n");
+};
+
+Migration.prototype.sqlFileLoaderIgnoreOnInitTemplate = function() {
+  return [
+    "'use strict';",
+    "",
+    "var dbm;",
+    "var type;",
+    "var seed;",
+    "var fs = require('fs');",
+    "var path = require('path');",
+    "var ignoreOnInit = false;",
+
+
+    "",
+    "/**",
+    "  * We receive the dbmigrate dependency from dbmigrate initially.",
+    "  * This enables us to not have to rely on NODE_PATH.",
+    "  */",
+    "exports.setup = function(options, seedLink) {",
+    "  dbm = options.dbmigrate;",
+    "  ignoreOnInit = dbm.getInstance().internals.argv['ignore-on-init']",
```

Comment:
Hi Tobias,

Thanks for your comment,

I totally agree with you, I changed it.



On Wed, Sep 2, 2015 at 6:45 PM, Tobias Gurtzick <notifications@github.com>
wrote:

> In lib/migration.js
> <https://github.com/db-migrate/node-db-migrate/pull/301#discussion_r38548393>
> :
>
> > +    "var dbm;",
> > +    "var type;",
> > +    "var seed;",
> > +    "var fs = require('fs');",
> > +    "var path = require('path');",
> > +    "var ignoreOnInit = false;",
> > +
> > +
> > +    "",
> > +    "/**",
> > +    "  * We receive the dbmigrate dependency from dbmigrate initially.",
> > +    "  * This enables us to not have to rely on NODE_PATH.",
> > +    "  */",
> > +    "exports.setup = function(options, seedLink) {",
> > +    "  dbm = options.dbmigrate;",
> > +    "  ignoreOnInit = dbm.getInstance().internals.argv['ignore-on-init']",
>
> Actually this is not a good way to implement this. Creating a new instance
> of the whole migration framework within a migration is really to much. You
> can store the ignore-on.init within the options parameter which gets
> supplied here. See
> https://github.com/db-migrate/node-db-migrate/blob/master/api.js#L67
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/db-migrate/node-db-migrate/pull/301/files#r38548393>.
>